### PR TITLE
feat(lib, playwright): Automatically update validation files

### PR DESCRIPTION
**Pull request by foxable**
This PR adds extended support for Playwright's `updateSnapshot` setting. In addition to disabling retries during updates, validation files will now be automatically overridden when the update mode is `all` or `changed`. This significantly reduces the effort to update validation files, especially in larger test projects. The `output` files are always written, preserving the capability to manually update validation files using your preferred diffing tool.

Support for the update mode `none` is also added. It can be used to ensure that no validation files are written, e.g. in CI pipelines.

### CODE REVIEW

File: packages/playwright-file-snapshots/tests/update-snapshot.spec.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  expect(readSnapshotFile(snapshotDirs.validationDir)).toStrictEqual([
+    "===== missing file =====",
+    "initial value",
+    "",
+  ]);
+});
+
+test(
+  "when snapshot updates are enabled, creates missing validation file",
+  { tag: [SNAPSHOT_UPDATE_TAG] },
```
[the remark this answers is not in the document]
Comment (foxable) [reply packages/playwright-file-snapshots/tests/update-snapshot.spec.ts]: I don't think so. At first I thought it would be hard to configure them for test purposes, but it should be possible to add another test call using `--update-snapshots none`. I will try it out.

File: packages/playwright-file-snapshots/tests/update-snapshot.spec.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+    const snapshotDirs = temporarySnapshotDirs();
+    const testExpect = defineValidationFileExpect(snapshotDirs);
+
+    await expect(() =>
+      testExpect("initial value").toMatchTextFile(),
+    ).rejects.toThrowError();
+    await expect(() =>
+      testExpect("changed value").toMatchTextFile(),
+    ).rejects.toThrowError();
```
[the remark this answers is not in the document]
Comment (foxable) [reply packages/playwright-file-snapshots/tests/update-snapshot.spec.ts]: I like the idea to test that another matcher call actually succeeds, because the validation file is updated. But I think I prefer individual assertions, because they better describe the expected behavior and will be able to find regressions regarding this behavior.

Would you still keep the assertion on the final content of the validation file? Because strictly speaking, a non-failing last matcher call already asserts that the validation file was updated to the new value.

File: packages/lib-file-snapshots/src/matcher/validation-file-matcher.test.ts [hunk 3]
```diff
@@ -93,10 +121,95 @@ test("when validation file exists, does not recreate validation file", async (co
...
+  expect(matcher.isValidationFileMissing).toBe(false);
+  expect(matcher.isUpdate).toBe(true);
+
+  const changedMatcherResult = matcher.matchFileSnapshot("changed value");
+  await snapshotMatcherResult(context, changedMatcherResult, tmpDir);
+  changedMatcherResult.writeFileSnapshots();
+
+  expect(readSnapshotFile(changedMatcherResult.validationFilePath)).toBe(
+    addTrailingNewLine("changed value"),
+  );
```
[the remark this answers is not in the document]
Comment (foxable) [reply packages/lib-file-snapshots/src/matcher/validation-file-matcher.test.ts]: Alright, that's good feedback. I added it to my refactoring backlog and will tackle it in a follow-up. I'm already planning to rework some of the tests.

File: packages/playwright-file-snapshots/tests/update-snapshot.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (foxable) [reply packages/playwright-file-snapshots/tests/update-snapshot.spec.ts]: Update: I tried it out, but there are cases where I need the explicit assertion on the file contents (e.g. when creating the validation file, because the missing file header will never yield succeeding matches). So I need the custom assertion either way. That's why I decided against removing it for the other tests to keep the tests consistent. Instead, I extracted functions and constants to make the tests more readable.

File: packages/playwright-file-snapshots/tests/update-snapshot.spec.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  ).rejects.toThrowError();
+}
+
+async function matchChangedValue(
+  testExpect: Expect<PlaywrightValidationFileMatchers>,
+): Promise<void> {
+  await expect(() =>
+    testExpect("changed value").toMatchTextFile(),
+  ).rejects.toThrowError();
+}
```
[the remark this answers is not in the document]
Comment (foxable) [reply packages/playwright-file-snapshots/tests/update-snapshot.spec.ts]: You have a point here. I renamed the functions to  `matchInitialValueWithError` and `matchChangedValueWithError` to clarify that they try to match the value but expect a matching error.
